cesql/spec.md

```diff
@@ -278,6 +278,10 @@ For example, the pattern `_b*` will accept values `ab`, `abc`, `abcd1` but won't
 Both `%` and `_` can be escaped with `\`, in order to be matched literally. For example, the pattern `abc\%` will match
 `abc%` but won't match `abcd`.
 
+In cases where the left operand is not a `String`, it MUST be cast to a `String` before the comparison is made. 
+The pattern of the `LIKE` operator (that is, the right operand of the operator) MUST be a valid string predicate, 
+otherwise the parse MUST return a parse error.
```

Comment:
Just to be clear... `subject LIKE TRUE` isn't valid because `TRUE` isn't a valid string, however, we talk about how there's implicit casting all over the place... so someone may wonder why `TRUE` isn't implicitly converted into `"true"`. Am I correct in this thinking? Should we be explicit and say that casting isn't allowed in this case? Or perhaps should we allow it for consistency?  @jskeet thoughts?

Reply:
It would definitely be good to be clear. Even without the "LIKE" aspect, I can't tell offhand whether `"TRUE" = true` is casting the LHS to Boolean, or the RHS to String. I suspect it's the former, based on bullet 2 in the list in 3.7.

Reply:
@duglin I think the main problem is that currently the `LIKE` expression isn't defined with an expression as the right operand. It is defined as:
```ebnf
expression NOT? LIKE stringLiteral
```

So, any value on the right operand which is not a string literal MUST be a parse error currently. We can change this, but I'm not sure if it makes sense. The whole point of a `LIKE` expression is to compare against a pattern, not a specific value. If someone were to compare against another string value wouldn't it just makes sense to use `=` or something like that?

Reply:
I think it's ok to leave it as "must be a string literal w/o casting", but let's be sure clear in the text that implicit type casting isn't allowed in this one spot. I agree defining it as `stringLiteral` sort of implies it, but saying that things should be implicitly case in all other spaces (even when the spec says the arg is a `string`) could be unclear to people.